### DISCUSSION TITLE
Lost my apps when upgrading ???

### DISCUSSION THREAD

**Discussion by kmdelucaNG** [Budibase/budibase#4249] [Help]
Hello,
Per https://docs.budibase.com/self-hosting/self-host#updating-budibase, used `budi hosting --update ` to upgrade from v0.9.168 to v1.0.46.
Upgrade went fine - except now my Apps are gone.  Everything else seems to be there (Organization, Users).
Did I need to save them off and then import them or something?

**Comment by kmdelucaNG** [comment]
Any ideas on this from the community?   I'm about to try another upgrade. Thanks.

**Reply by Kuuchuu** [reply]
Are you using the Docker install? If so make sure your docker-compose matches the new version, there have been some big changes between those versions.
If you're using Docker, personally I would downgrade your Budibase install first, backup all apps if they are once again accessible, then do the upgrade again and ensure your compose file is up-to-date.

**Reply by kmdelucaNG** [reply]
yes, using docker install.  And I had a VM snapshot - so I ended up reverting back to v0.9.168 with apps in tact.  I'm going to try exporting the apps and seeing/hoping if they are imported after an upgrade. thanks.

**Comment by gerhardmpl** [comment]
@kmdelucaNG I always update using `budi hosting --update` and never lost an app (I started with the V1.x version). But I always do a Budibase VM snapshot and an export of important apps before updating, just to be sure.

**Reply by kmdelucaNG** [reply]
yes, I had a VM snapshot that I was able to recover with.  And I have also done an App export now in prep for the next upgrade attempt - I had just never experienced this problem before.  But as @Kuuchuu says above - maybe this is a symptom of trying to perform a "too drastic" version upgrade.  I'll be trying again soon. thanks.

**Comment by kmdelucaNG** [comment]
Update seemed to go OK... and EVENTUALLY (after a minute or 2) the apps showed up in developer mode - so they may have been working previously too??? I do get an error saying "Error loading apps and templates" though. I can see/access the apps and templates - so maybe that's a false alarm???
![image](https://user-images.githubusercontent.com/86023779/165423561-fa1c9647-0a1f-47c6-95c2-8410ee50d67a.png)
